### PR TITLE
feat: api refactor

### PR DESCRIPTION
**Pull request by hwjiangkai**
Signed-off-by: jyjiangkai <jyjiangkai@163.com>

### CODE REVIEW

File: golang/api.go [hunk 1]
```diff
@@ -15,17 +15,62 @@
...
+	Disconnect() error
+}
+
+type Publisher interface {
+	Eventbus() string
+	Publish(ctx context.Context, events ...*v2.Event) error
+}
+
+type Subscriber interface {
+	Subscription() string
```
Comment (wenfengwang) [inline golang/api.go]: SubscriptionID or what?

File: golang/api.go [hunk 2]
```diff
@@ -15,17 +15,62 @@
...
+	SubscriptionID string
+}
+
+type Controller interface {
+	Eventbus(name string) Eventbus
+	Subscription(id string) Subscription
+}
+
+type Eventbus interface {
+	List() ([]*metapb.EventBus, error)
```
Comment (wenfengwang) [inline golang/api.go:61]: use a wrapper object?

File: golang/api.go [hunk 3]
```diff
@@ -15,17 +15,62 @@
...
+	Get() (*metapb.EventBus, error)
+	Create() error
+	Delete() error
+	Update() error
```
Comment (wenfengwang) [inline golang/api.go]: remove this method this version?

File: golang/api.go [hunk 4]
```diff
@@ -15,17 +15,62 @@
...
+}
+
+type Subscription interface {
+	List() ([]*metapb.Subscription, error)
+	Get() (*metapb.Subscription, error)
+	Create() error
+	Delete() error
+	Update() error
```
Comment (wenfengwang) [inline golang/api.go]: ditto

File: golang/api.go [hunk 5]
```diff
@@ -15,17 +15,62 @@
...
+	Disable() error
```
Comment (wenfengwang) [inline golang/api.go]: is `pause` better?

File: golang/client.go [hunk 6]
```diff
@@ -2,181 +2,62 @@ package golang
...
-	close(sc.messagec)
-	sc.state = stateClosed
-}
-
 type client struct {
-	Endpoint    string
-	proxy       vanuspb.ClientClient
-	streamCache sync.Map
-	mu          sync.Mutex
+	Endpoint        string
```
Comment (wenfengwang) [inline golang/client.go]: why this field is exported?

File: golang/controller.go [hunk 7]
```diff
@@ -0,0 +1,79 @@
+package golang
```
Comment (wenfengwang) [inline golang/controller.go:15]: add Header license for all source code files

File: golang/subscriber.go [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+import (
+	"context"
+	"sync"
+
+	v2 "github.com/cloudevents/sdk-go/v2"
+	"github.com/linkall-labs/vanus/observability/log"
+	proxypb "github.com/linkall-labs/vanus/proto/pkg/proxy"
+)
+
+type ackCallback func(result bool) error
```
Comment (wenfengwang) [inline golang/subscriber.go:26]: can use the enum to represent result?

Comment (wenfengwang) [reply golang/subscriber.go:26]: or is this necessary?

**Review by wenfengwang — CHANGES_REQUESTED** [review]

File: golang/subscriber.go (hunk 8 above)
Comment (hwjiangkai) [reply golang/subscriber.go:26]: optimize in a separate pr and remains unchanged in this pr.

File: golang/api.go (hunk 2 above)
Comment (hwjiangkai) [reply golang/api.go:61]: optimize in a separate pr and remains unchanged in this pr.

**Review by wenfengwang — APPROVED** [review]
LGTM

**Pull request merged**